### PR TITLE
Retry Key Validation in Sparse Server Failure Case

### PR DESCRIPTION
**Pull request by Jacobsolawetz**
# Description

Retry auth key validation in the event of a sparsely occurring server authentication failure

### CODE REVIEW

File: roboflow/__init__.py [hunk 1]
```diff
@@ -28,11 +29,23 @@ def check_key(api_key, model, notebook):
...
+                    print("retrying...")
+                    time.sleep(1)
+                    num_retries += 1
+                    return check_key(api_key, model, notebook, num_retries)
+                else:
+                    raise RuntimeError(
+                        "There was an error validating the api key with Roboflow server."
+                    )
             else:
+                r = response.json()
```
Comment (yeldarby) [inline roboflow/__init__.py:48]: Should we put a try/catch around this & retry if it fails to produce valid JSON (but wit ha 200 error code) too?

Comment (yeldarby) [reply roboflow/__init__.py:48]: We should prob do this retry logic around all API routes not just the initialization

Comment (Jacobsolawetz) [reply roboflow/__init__.py:48]: Thought about that... I think this is safe if we get a 200 code, it will always produce valid JSON

Comment (Jacobsolawetz) [reply roboflow/__init__.py:48]: Interesting on _all_ routes - before undertaking that, do we think all api routes have this same vulnerability?

**Review by FrancescoSaverioZuppichini — APPROVED** [review]
LGTM! :rocket:

**Pull request merged**